docs/detections/rules-ui-create.asciidoc

```diff
@@ -263,35 +263,76 @@ For example, if a rule has an interval of 5 minutes, no additional look-back tim
 [[create-esql-rule]]
 === Create an ES|QL rule
 
-IMPORTANT: This is a placeholder for future documentation. The following content is incomplete.
+preview::[]
 
-. Go to *Rules* -> *Detection rules (SIEM)* -> *Create new rule*. The *Create new rule* page displays.
-. To create a rule that uses ES|QL, select **ES|QL**, 
-then write a query. There are two types of ES|QL queries:
+Use {ref}/esql.html[ES|QL] to query your source events and aggregate event data. Query results are returned in a table with rows and columns. Each row becomes an alert.
+
+To create an ES|QL rule:
 
-.. **Aggregating query**: This is a query that uses the `STATS...BY` grouping commands. Query results cannot be matched with a particular document in Elasticsearch. For example:
+. Go to *Rules* -> *Detection rules (SIEM)* -> *Create new rule*. The *Create new rule* page displays.
+. Select **ES|QL**, then write a <<esql-rule-query-types,query>>. 
 +
-[esql]
------
+TIP: Click the help icon (image:images/esql-ref-button.png[Click the ES|QL help icon,20,20]) to open the in-product reference documentation for all ES|QL commands and functions.
++
+. Click *Continue* to <<rule-ui-basic-params, configure basic rule settings>>.
+
+[float]
+[[esql-rule-query-types]]
+==== ES|QL query types 
+
+ES|QL rule queries are loosely categorized into two types: <<esql-agg-query,aggregating>> and <<esql-non-agg-query,non-aggregating>>. 
+
+[float]
+[[esql-agg-query]]
+===== Aggregating query 
+
+An aggregating query uses {ref}/esql-functions-operators.html#esql-agg-functions[`STATS...BY`] functions to aggregate source event data. Generated alerts only contain the fields returned by the query.
+
+Here is an example aggregating query:
+
+[source,esql]
+----
 FROM logs*
-| STATS count = COUNT(host.name) BY host.name
-| SORT host.name
+| STATS host_count = COUNT(host.name) BY host.name
+| SORT host_count DESC
+| WHERE host_count > 20
+----
+
+This query generates an alert whenever the `host.name` field is encountered more than 20 times during the specified rule interval.
+
+[float]
+[[esql-non-agg-query]]
+===== Non-aggregating query 
+A non-aggregating query doesn't use `STATS...BY` functions and doesn't aggregate source data. Generated alerts contain source event data and fields returned by the ES|QL query. 
+
+If you create a non-aggregating query, you also have to manually turn on alert deduplication and link source events to alerts. To do this, you must add the `[metadata _id, _index, _version]` operator after the `FROM` source command. For example:  
+
+[source,esql]
+-----
+FROM logs* [metadata _id, _index, _version]
+| WHERE event.id == "8a4f500d"
+| LIMIT 10
```

Comment:
Just realised, LIMIT over rule max_signals won't have any effect on number of generated alerts. It will always pick smaller between:
`LIMIT` in ES|QL query and rule [max_signals](https://www.elastic.co/guide/en/security/current/rules-api-create.html#opt-fields-all) in rule's configuration

Reply:
This is a nice catch, and a good "gotcha" to note in the rule docs. This information applies to ES|QL rule queries in general, so it might make sense to place it under step 2 [here](https://github.com/elastic/security-docs/pull/4062/files#diff-bb6609d5341af025e47b175f8432a1ce9dd90d3c93d656dbc388c599b120924dR273). In the next iteration of these docs, maybe we can move it to a new section/page that provides additional query guidance and examples. 

It might also be a good idea to mention this information about the `LIMIT` and `max_signals` params in the `max_signals` [API docs](https://www.elastic.co/guide/en/security/current/rules-api-create.html#opt-fields-all). Just so there's coverage in both places and users see if whether they create an ES|QL rule from the UI or API. What do you think, @vitaliidm ?

Reply:
Let's move it to step 2.

> It might also be a good idea to mention this information about the LIMIT and max_signals params in the max_signals [API docs](https://www.elastic.co/guide/en/security/current/rules-api-create.html#opt-fields-all). Just so there's coverage in both places and users see if whether they create an ES|QL rule from the UI or API. What do you think, @vitaliidm ?

I don't think we need to add it there. It's only related to ES|QL query.

Reply:
Quick update: requested language from Vitalii. Am keeping this thread open in the meantime.

Reply:
Chatted with Vitalii over Slack, and will need to add the following information:
- LIMIT defines how many rows will be returned, and because of this, the max number alerts per rule execution (`max_signals`)cannot be greater than what's defined in LIMIT.
- If LIMIT isn't present, `max_signals` will be taken in account.

Reply:
Added in [2d45097](https://github.com/elastic/security-docs/pull/4062/commits/2d450970274daa8dedf92385092f1d24f825615e).